msticpy/sectools/geoip.py

```diff
@@ -278,27 +293,139 @@ class GeoLiteLookup(GeoIpLookup):
-    _db_age = datetime.utcnow() - _last_mod_time
-    if _db_age > timedelta(40):
-        print(
-            f'{_DB_FILE} is over one month old. Update the maxminddb package')
-        print(
-            f'to refresh or download a new version from {_MAXMIND_DOWNLOAD}')
-
-    def __init__(self):
-        """Return new instance of GeoLiteLookup class."""
-        self._reader = geolite2.reader()
-
-    def lookup_ip(self, ip_address: str = None,
-                  ip_addr_list: Iterable = None,
-                  ip_entity: IpAddress = None) -> Tuple[List[Tuple[Dict[str, str], int]],
-                                                        List[IpAddress]]:
+    _MAXMIND_DOWNLOAD = (
+        "https://geolite.maxmind.com/download/geoip/database/GeoLite2-City.mmdb.gz"
+    )
+    _PKG_DIR = os.path.join(os.path.dirname(site.__file__), "site-packages", "msticpy")
+    _DB_ARCHIVE = "GeoLite2-City.mmdb.gz"
+    _DB_FILE = "GeoLite2-City.mmdb"
+
+    def download_and_extract_gzip(self, url: str = None, db_folder: str = None):
+        """Helper function to download file from the given URL and extract if it is archive
+
+        Parameters
+        ----------
+        url : str
+            Web URL location to the Maxmind city Database. (the default is None)
+        db_folder: str, optional
+            Provide absolute path to the folder containing MMDB file (e.g. '/usr/home' or 'C:\maxmind'). 
+            If no path provided, it is set to download msticpy package installation directory.(the default is None)
+        """
+
+        if url is None:
+            url = self._MAXMIND_DOWNLOAD
+
+        if db_folder is None:
+            db_folder = self._PKG_DIR
+
+        _DB_ARCHIVE_PATH = os.path.join(db_folder, self._DB_ARCHIVE)
+        _DB_FILE_PATH = os.path.join(db_folder, self._DB_FILE)
+
+        try:
+            response = requests.get(url, stream=True)
+            response.raise_for_status()
+        except HTTPError as http_err:
+            print(f"HTTP error occurred: {http_err}")
+        except Exception as err:
+            print(f"Other error occurred: {err}")
+        else:
+            print("Downloading GeoLite DB archive from MaxMind....")
+            with open(_DB_ARCHIVE_PATH, "wb") as fd:
+                for chunk in response.iter_content(chunk_size=10000):
+                    fd.write(chunk)
+                print(f"Downloaded archive location :: {_DB_ARCHIVE_PATH}")
+            try:
+                with gzip.open(_DB_ARCHIVE_PATH, "rb") as f_in:
+                    print(f"Extracting city database...")
+                    with open(_DB_FILE_PATH, "wb") as f_out:
+                        shutil.copyfileobj(f_in, f_out)
+                        print(
+                            f"Extraction complete. Local Maxmind city DB :: {_DB_FILE_PATH}"
+                        )
+            except IOError as err:
+                print(f"{_DB_ARCHIVE_PATH} {err}")
+
+    def get_geolite_dbpath(
```

Comment:
Split into 
1. get_geolite_dbpath property method (@property decorator) that returns the current path
2. _check_and_update_version (or some appropriate name)

Reply:
need to check about decorator in the current modified code. have added new function _check_and_update.